python/ray/train/_internal/data_config.py

```diff
@@ -68,12 +76,19 @@ def configure(
             equal to `world_size`. Each element of the list contains the assigned
             `DataIterator` instances by name for the worker.
         """
-        output = [{} for i in range(world_size)]
+        output = [{} for _ in range(world_size)]
+
+        if self._datasets_to_split == "none":
+            datasets_to_split = set()
+        elif self._datasets_to_split == "all":
+            datasets_to_split = set(datasets.keys())
+        else:
+            datasets_to_split = set(self._datasets_to_split)
 
         for name, ds in datasets.items():
```

Comment:
let's also double check there's no `dataset` has `name == "all"`, if that's the case, we should throw an exception to avoid surprise behavior for users.

Reply:
Oh actually the type of `datasets_to_split` is either a string literal "all", or a list of str. So theoretically, if they have a dataset named "all", they should set `DataConfig(datasets_to_split=["all"])` to split it. If they want to split all, they should set `DataConfig(datasets_to_split="all")`

Do you think it's good to forbid the dataset name "all" here?

Reply:
sorry I miss the `["all"]` part. Then I think "all" is good now.

Reply:
shall I move the checks from the "__init__" function here?